pkg/controller/bundle_sync_task.go

```diff
@@ -70,19 +71,24 @@ func (st *bundleSyncTask) process() (retriableError bool, e error) {
 			processedResources: st.processedResources,
 			plugins:            st.plugins,
 			scheme:             st.scheme,
+			blockedOnError:     blockedOnError,
 		}
 		resInfo := rst.processResource(&res)
 		if retriable, err := resInfo.fetchError(); err != nil && api_errors.IsConflict(errors.Cause(err)) {
 			// Short circuit on conflict
 			return retriable, err
 		}
-		log.Printf("[WORKER][%s/%s] Resource %q, ready: %t", st.bundle.Namespace, st.bundle.Name, resName, resInfo.isReady())
+		_, resErr := resInfo.fetchError()
+		blockedOnError = blockedOnError || resErr != nil
+		log.Printf("[WORKER][%s/%s] Resource %q, ready: %t, error: %v", st.bundle.Namespace, st.bundle.Name, resName, resInfo.isReady(), resErr)
 		st.processedResources[resName.(smith_v1.ResourceName)] = &resInfo
 	}
-	// Delete objects which were removed from the bundle
-	retriable, err := st.deleteRemovedResources()
-	if err != nil {
-		return retriable, err
+	if !blockedOnError {
```

Comment:
I'd be tempted to just have a method which traverses processResources here rather than cache the state. Should be cheap? Or am I misunderstanding something?
---
Also, I'm trying to understand the exact condition under which objects are removed from the bundle. Shouldn't this object removal only take place once the bundle has completely finished? At the moment, it looks like it deletes after the initial pass, when some resources are still blocked/in progress? If this is like CF, this is not a good idea (for example, the old compute could still be up referring to a particular resource...)

Reply:
Why you think a method would be better?

> Shouldn't this object removal only take place once the bundle has completely finished?
This is bug #77, I know about this. I don't want to fix it without a test and this PR is too big already.

Reply:
I just don't like caching when it's not necessary (possibility of inconsistent state, etc.). When you move the handling of object removal out of here, there will be nothing so close to this traversal anyway.

Reply:
This flag is also passed to `resourceSyncTask` which needs it. I.e. it is needed during processing, not only after it. I don't see why using a method in this case is better in any way (I would probably agree about iteration at the end otherwise).